header/p2p/server_test.go

```diff
@@ -0,0 +1,27 @@
+package p2p
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-datastore"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/header/store"
+)
+
+func TestExchangeServer_handleRequestTimeout(t *testing.T) {
+	_, peer := createMocknet(t)
+	s, err := store.NewStore(datastore.NewMapDatastore())
+	assert.NoError(t, err)
+	server := NewExchangeServer(peer, s, "private")
+	err = server.Start(context.Background())
+	assert.NoError(t, err)
+	t.Cleanup(func() {
+		server.Stop(context.Background()) //nolint:errcheck
+	})
+
+	_, err = server.handleRequest(1, 200)
```

Comment:
so will this hang for 1 minute? where does the ctx cancellation happen?

Reply:
https://github.com/celestiaorg/celestia-node/pull/1318#discussion_r1019016683

Reply:
> where does the ctx cancellation happen?

https://github.com/vgonkivs/celestia-node/blob/b06aa8666feb25098d1af582546ff0356d7ee0e4/header/store/heightsub.go#L64

Reply:
I see now that it's 5 seconds.